src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageContextExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Text;
+using Arcus.Messaging.Abstractions;
+using GuardNet;
+
+namespace Arcus.Messaging.ServiceBus.Core.Extensions
+{
+    /// <summary>
+    /// Extensions on the <see cref="MessageContext"/> to make accessing values more dev-friendly.
+    /// </summary>
+    public static class MessageContextExtensions
+    {
+        /// <summary>
+        /// Gets the encoding property value in the messaging context, using UTF-8 as default if no such encoding value can be retrieved.
+        /// </summary>
+        /// <param name="messageContext">The context of the message.</param>
+        public static Encoding GetMessageEncodingProperty(this MessageContext messageContext)
+        {
+            Guard.NotNull(messageContext, nameof(messageContext));
+
+            if (messageContext.Properties.TryGetValue(PropertyNames.Encoding, out object annotatedEncoding))
+            {
+                try
+                {
+                    return Encoding.GetEncoding(annotatedEncoding.ToString());
+                }
+                catch
+                {
+                    // Ignore.
```

Comment:
Thought - Should we log it? Because something was specified but we couldn't parse it so we'll use UTF 8 instead.

Migth be easier for troubleshooting

Reply:
Hmm, then we have to pass in the logger.

Reply:
I wouldn't make it optional, otherwise you are in the dark

Reply:
Nevermind, it's ok